node/internal.ts

```diff
@@ -442,10 +442,10 @@ function _tryGetExecutablePath(filePath: string, extensions: string[]): string {
                 }
             }
             else {
-                // on Mac/Linux, test the execute bit
+                // on Mac/Linux, test the execute bits
                 //     R   W  X  R  W X R W X
                 //   256 128 64 32 16 8 4 2 1
-                if ((stats.mode & 1) == 1) {
+                if ((stats.mode & 1) > 0 || ((stats.mode & 8) > 0 && stats.gid === process.getgid()) || ((stats.mode & 64) > 0 && stats.uid === process.getuid())) {
```

Comment:
would a private helper method be better since this is used from two locations (single unit test would always excersize the same code)

Reply:
Yep, good call